mongo/integration/retryable_reads_prose_test.go

```diff
@@ -36,6 +37,10 @@ func TestRetryableReadsProse(t *testing.T) {
 	defer mt.Close()
 
 	mt.Run("PoolClearedError retryability", func(mt *mtest.T) {
+		if os.Getenv("SERVERLESS") == "serverless" {
+			mt.Skip("skipping as serverless has different pool clearing behavior")
+		}
```

Comment:
```suggestion
		if mtest.ClusterTopologyKind() == mtest.LoadBalanced {
			mt.Skip("skipping as load balanced topology has different pool clearing behavior")
		}
```

I think the cause of the pool clearing behavior is due to serverless using a load balancer. The LoadBalanced topology has [different pool clearing behavior](https://github.com/mongodb/specifications/blob/4e17d5700d17eb7663acdaa392fc08165f422af9/source/load-balancers/load-balancers.rst#initial-handshake-errors).

Currently, `TestRetryableReadsProse` is [not run in the load balancer tests](https://github.com/mongodb/mongo-go-driver/blob/a80ae1d32c8973141bea60d097eb00d86e772ede/Makefile#L170). If it does run, it would need to skip as well.

Reply:
Ah great point; we're really skipping because of the load balancer pool clearing behavior.